src/link/patch.c

```diff
@@ -60,18 +60,22 @@ static int32_t asl(int32_t value, int32_t shiftamt)
 	return (uint32_t)value << shiftamt;
 }
 
-/* This is an "empty"-type stack */
+/* This is an "empty"-type stack. Apart from the actual values, we also remember
+ * whether the value is a placeholder inserted for error recovery. This allows
+ * us to avoid cascading errors. */
 struct RPNStack {
-	int32_t *buf;
+	int32_t *values;
+	bool *errorFlags;
```

Comment:
Why is this an array?

Reply:
Isn't it one per value?

Reply:
Yes, has the commentary not made that clear?

Reply:
But why do we need to remember it for every single "parent" value, and not the current one?

Reply:
The intent is to report as much as possible, such as `ld hl, UnknownSymbol / ExpressionThatHappensToBeZero`

Reply:
But aren't those reported as soon as they're encountered?

Reply:
If we don't remember whether it is the left or right element of the division that had the error, there is no way to determine, short of guessing, whether we should report a division-by-zero error.

Reply:
Ah, I think I understand what seems to be the misunderstanding. The values in `errorFlags` correspond 1:1 to `values`, not to positions in the RPN command stream. In other words, if we consider the AST, we store the error flags for the nodes whose parents haven't been evaluated yet.

Reply:
The fact that only one index is ever used out of the entire `errorFlags` array makes me unable to shake off the feeling that this can be implemented using a single `bool` instead of an array... and I don't want to merge until I understand why not :/

Reply:
The index used with `errorFlags` is always the same as the one used with `values`. Therefore, you should replace `values` with only a single `int32_t` ;)

I'm not sure anymore whether it wouldn't be better to wrap this in a `struct StackValue { int32_t value; bool isError; };`. I believe the large overhead of padding made me prefer a struct-of-arrays approach, though I can change this if you think readability is a more important factor here.